staging/src/k8s.io/apiserver/pkg/server/config_selfclient.go

```diff
@@ -145,7 +145,7 @@ func (s *ServingInfo) NewSelfClientConfig(token string) (*restclient.Config, err
 		return nil, nil
 	}
 	return &restclient.Config{
-		Host: s.BindAddress,
+		Host: "http://" + s.BindAddress,
```

Comment:
Or even better put it into a small helper `func (s *ServingInfo) loopbackHost() string` and use that at both places.